### ISSUE TITLE
CardinalityEstimator.Merge() Keeps Direct Counting Items No Matter How Many Exist After The Merge

### ISSUE THREAD

**Issue by J-Griffin** [saguiitay/CardinalityEstimation#23]
I've run into a case where the CardinalityEstimator serialized size grows larger and larger because the direct counted items list is never cleared. 

We have a set of estimators which are later merged together (no additional Adds are done). If the estimators are below 100 items, they will hold 100+ items after the merge. If more small estimators are later merged in, the direct count list will continue to grow.

In this example, I'm using serialize to demonstrate the increasing size of the "mergedEstimator" vs the add-only estimator. If the loop is increased from 10k to 100k, the mergedEstimator continues to increase in size:

```c#
var addedEstimator = new CardinalityEstimator();
var mergedEstimator = new CardinalityEstimator();

for(int i = 0; i < 10_000; i++) {
	var guid = Guid.NewGuid().ToString();
	
	addedEstimator.Add(guid);

	// Simulate some intermediate estimators being merged together
	var temporaryEstimator = new CardinalityEstimator();
	temporaryEstimator.Add(guid);
	mergedEstimator.Merge(temporaryEstimator);
}

var serializer = new CardinalityEstimatorSerializer();

var stream1 = new MemoryStream();
serializer.Serialize(stream1, addedEstimator, true);
Console.WriteLine($"Size of 'Just added' estimator: {stream1.Length.ToString("N0")}");

var stream2 = new MemoryStream();
serializer.Serialize(stream2, mergedEstimator, true);
Console.WriteLine($"Size of merged estimator: {stream2.Length.ToString("N0")}");
```

Output:
> Size of 'Just added' estimator: 16,406
> Size of merged estimator: 80,022

**Comment by saguiitay** [comment]
I'm handling this in #26 which takes your change, and also includes a UT.

**Comment by J-Griffin** [comment]
I've tested this with version 1.8.0 in nuget, but it still seems to happen. It looks like 1.8.0 doesn't have the new change. I see this from ILSpy:

```c#
public void Merge(CardinalityEstimator other)
{
	// ...snip...
	if (other.directCount != null)
	{
		if (directCount != null)
		{
			directCount.UnionWith(other.directCount);
		}
	}
	else
	{
		directCount = null;
	}
}
```

@saguiitay can you verify nuget deploy got the right bits?

**Comment by J-Griffin** [comment]
I DO see the fix in nuget v1.9.0. Thanks!
